chunky/src/java/se/llbit/chunky/resources/BitmapImage.java

```diff
@@ -176,4 +171,32 @@ public static BitmapImage concatY(BitmapImage a, BitmapImage b) {
     img.blit(b, 0, a.height, 0, 0, b.width, b.height);
     return img;
   }
+
+  @Deprecated
+  public void setData(int[] image) {
+    // Check for size mismatch
+    if (image.length<width*height)
+      throw new IllegalArgumentException("Size not matched. Expected "
+          +width+"*"+height+"="+(width*height)+" but got "+image.length);
+
+    for (int row = 0; row < height; row++)
+      System.arraycopy(image, width * row, data[row], 0, width);
+  }
+
+  // This should ONLY BE USED for smaller textures, and not rendering the full output
+  // (eg Scene.backBuffer), where this will fail.
+  @Deprecated
+  public int[] toIntArray() {
+    if (((long)width)*height>Integer.MAX_VALUE)
+      throw new IndexOutOfBoundsException("Bitmap too large. Cannot fit within a single array. This should only be used for small images, such as textures or MapTiles, not full renders!");
+    int[] copy = new int[width*height];
+    for (int row = 0; row<height; row++)
+      System.arraycopy(data[row],0,copy,width*row,width);
+    return copy;
+  }
+
+  @Deprecated
+  public int[] getRow_Unsafe(int row) {
+    return data[row];
+  }
```

Comment:
Are these methods used anywhere in Chunky? I can update the denoiser if that's the only issue.

Reply:
toIntArray() is used in `MapTile#drawImage(BitmapImage)`
toIntArray() is used in `FxImageUtil#toFxImage(BitmapImage)`

getRow_Unsafe() is used in `RenderCanvasFx#forceRepaint(BitmapImage)`